hathor/transaction/transaction.py

```diff
@@ -349,10 +355,10 @@ def _update_token_info_from_outputs(self, *, token_dict: dict[TokenUid, TokenInf
                     sum_tokens = token_info.amount + tx_output.value
                     token_dict[token_uid] = TokenInfo(sum_tokens, token_info.can_mint, token_info.can_melt)
 
-    def iter_spent_rewards(self) -> Iterator[Block]:
+    def iter_spent_rewards(self, storage: Optional['SimpleMemoryStorage'] = None) -> Iterator[Block]:
```

Comment:
It seems like we are juggling to make the new memory storage work alongside the current code. I think we can spawn an ephemeral verifier that carries a storage and use it to run the verification. It can be either set to `tx.storage` or to any other storage. What do you think?

Reply:
Indeed, this was not a good solution. I have solved this using a `StorageProtocol`, so both `TransacionStorage` and `SimpleMemoryStorage` have an unified interface that can be received by those methods. It was implemented in https://github.com/HathorNetwork/hathor-core/pull/922, and after it's merged to master I'll rebase this PR so this is updated.